crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -2024,6 +2046,19 @@ impl AuthorityPerEpochStore {
             .epoch_total_duration
             .set(self.epoch_open_time.elapsed().as_millis() as i64);
     }
+
+    // TODO: should be pub(crate) when it is inserted only from consensus
+    pub fn insert_oauth_jwk(&self, content: &OAuthProviderContent) {
+        if self.signature_verifier.insert_oauth_jwk(content) {
+            self.tables
```

Comment:
Could a data race be possible? We first pass this if check, then somewhere else we are using the sig to verify things, did some side effects, and before writing to the db below we crash. Then we have something verified without a materialized auth.
A more robust solution is probably just check existence in the if, insert to db first, before inserting to in-memory map?

Reply:
I think these concerns will go away once we are updating the JWKs via consensus